modules/otel-troubleshoot-instrumentation.adoc

```diff
@@ -0,0 +1,91 @@
+// Module included in the following assemblies:
+//
+// * observability/otel/otel-troubleshooting.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="troubleshooting-instrumentation_{context}"]
+= Troubleshooting instrumentation
+
+You can troubleshoot instrumentation issues as follows.
+
+.Procedure
+
+== Ensure the Instrumentation was injected properly in your workload
+
+. Check the instrumentation object is created:
++
+[source,console]
+----
+oc get instrumentation -n <your-project>
+----
+
+. Verify that the resources were deployed in the correct order
++
+Deploy the Instrumentation resource before the application to ensure auto-instrumentation works correctly.
+
+When the pod starts, the operator checks for the annotation to know if it needs to inject an instrumentation. It will check for an Instrumentation and inject auto-instrumentation into the pod based on the settings specified in the annotation. Generally, this process adds an init-container to the application’s pod, which is responsible for injecting the auto-instrumentation into the app container and some environment variables.
+
+If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.
+
+To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
```

Comment:
Is this all part of one step that is written on line 22?

Reply:
Yes but I don't have a strong opinion about moving it to its own "section". Maybe we can do something like:
```suggestion
Deploy the Instrumentation resource before the application to ensure auto-instrumentation works correctly.

When the pod starts, the operator checks for the annotation to know if it needs to inject an instrumentation. It will check for an Instrumentation and inject auto-instrumentation into the pod based on the settings specified in the annotation. Generally, this process adds an init-container to the application’s pod, which is responsible for injecting the auto-instrumentation into the app container and some environment variables.

If the Instrumentation resource is not available when the application is deployed, the operator will not inject the auto-instrumentation. When you want to update the instrumentation settings, you need to delete the instrumentation object and redeploy the application.

. Verify the instrumentation libraries were injected properly
To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:
```

Reply:
Re "To verify that the `opentelemetry-auto-instrumentation`` init-container has successfully started, execute the following command:"

@iblancasa, is this also applicable to Go?
 If not, then what should be here for Go?

Reply:
Yes, it is.

Reply:
@iblancasa, what is the `Instrumentation resource` on line 28? Is it an object or something else?

Reply:
Yes. It is another object provided by the OpenTelemetry Operator. Like the  OpenTelemetryCollector. We have a section called “Configuring Instrumentation” or something where you can find more info.

Reply:
Done.